src/app/components/course-list/course-list.component.html

```diff
@@ -1,19 +1,36 @@
-<h1 class="listTitle">Courses</h1>
+<h1 class="list__title">Courses</h1>
 
-<div class="listItemsWrapper">
-  <div><input placeholder="Text to search" /> <button>Search</button></div>
+<div class="list__itemsWrapper">
+  <div>
+    <input
+      placeholder="Text to search"
+      #searchInput
+      (input)="setSearchText(searchInput.value)"
+    />
+    <button (click)="setShouldFilterData()">Search</button>
+  </div>
 
```

Comment:
First, a lot of logic for course list component. Regarding SOLID iour course list component should be responsible only for showing list f courses. Let's move search to the separate component,  and provide search logic there.